vault/plugin_catalog.go

```diff
@@ -214,6 +214,13 @@ func (c *PluginCatalog) cleanupExternalPlugin(name, id string) error {
 		}
 		c.logger.Debug("killed external plugin process", "name", name, "pid", pc.pid)
 	} else if len(extPlugin.connections) == 0 || pc.client.Exited() {
```

Comment:
Do we also want to close the connection right before L210 for the non-multiplexing case?

Reply:
Yup! You were one step ahead of me 😅 Fixed now